### PR TITLE
Feature: Manage scope

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/test/resources/features/Role.feature [hunk 1]
```diff
@@ -328,6 +328,77 @@ Feature: Role feature
...
+    And  I expect response resources length is "1"
+    And  I expect one resource contains "id" equals to "[role_id]"
+    And  I expect one resource contains "name" equals to "ROLE"
+    And  I expect one resource contains "isDirect" equals to "true" as "boolean"
+
+    # Dissociate scope and role
+    When I request "/roles/[role_id]/scopes/[scope_id]" with method "DELETE"
+    Then I expect "204" as status code
+
+    # Role shouldn't have group
```
Comment (justinehell) [inline src/test/resources/features/Role.feature:390]: Role shouldn't have scope ?

File: src/test/resources/features/Scope.feature [hunk 2]
```diff
@@ -0,0 +1,143 @@
+Feature: scope feature
```
Comment (justinehell) [inline src/test/resources/features/Scope.feature:1]: Feature: Scope feature

File: src/test/resources/features/Scope.feature [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  Scenario: Should return 400 on a duplicated scope creation
+    Given I initialize the admin user
+    And   I clean scope "TEST"
+
+    When I request "/scopes" with method "POST" with json
+      | key  | value |
+      | name | TEST  |
+    Then I expect "201" as status code
+
+    When  I request "/scopes" with method "POST" with json
```
Comment (justinehell) [inline src/test/resources/features/Scope.feature:24]:
```suggestion
    When I request "/scopes" with method "POST" with json
```

File: src/test/resources/features/Scope.feature [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    And  I extract resources from response
+    And  I expect response resources length is "0"
+
+    # Check current user has only one scope.
+    When I request "/users/me/scopes" with method "GET"
+    Then I expect "200" as status code
+    And  I extract resources from response
+    And  I expect response resources length is "0"
+
+    Scenario:
```
Comment (justinehell) [inline src/test/resources/features/Scope.feature:143]: remove ?

File: src/test/resources/features/Scope.feature [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    Then I expect "201" as status code
+
+    # Check scope test has one associated user
+    When I request "/scopes/[scope_id]/users" with method "GET"
+    Then I expect "200" as status code
+    And  I extract resources from response
+    And  I expect response resources length is "1"
+    And  I expect one resource contains "login" equals to "admin"
+
+    # Check current user has two scopes.
```
Comment (justinehell) [inline src/test/resources/features/Scope.feature:120]: remove end `.`

File: src/test/resources/features/Scope.feature [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    When I request "/scopes/[scope_id]/users/admin" with method "DELETE"
+    Then I expect "204" as status code
+
+    # Check scope test has no users
+    When I request "/scopes/[scope_id]/users" with method "GET"
+    Then I expect "200" as status code
+    And  I extract resources from response
+    And  I expect response resources length is "0"
+
+    # Check current user has only one scope.
```
Comment (justinehell) [inline src/test/resources/features/Scope.feature:137]: remove end `.`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/test/resources/features/Scope.feature (hunk 2 above)
Comment (Zorin95670) [reply src/test/resources/features/Scope.feature:1]: done

File: src/test/resources/features/Scope.feature (hunk 4 above)
Comment (Zorin95670) [reply src/test/resources/features/Scope.feature:143]: done

File: src/test/resources/features/Role.feature (hunk 1 above)
Comment (Zorin95670) [reply src/test/resources/features/Role.feature:390]: done

File: src/test/resources/features/Scope.feature (hunk 5 above)
Comment (Zorin95670) [reply src/test/resources/features/Scope.feature:120]: done

File: src/test/resources/features/Scope.feature (hunk 6 above)
Comment (Zorin95670) [reply src/test/resources/features/Scope.feature:137]: done

**Review by justinehell — APPROVED** [review]

**Pull request merged**